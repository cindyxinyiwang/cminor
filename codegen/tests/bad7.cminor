// return statement
x: boolean = true;

a: function integer () = {
	arr: array [12] integer;

	return x;
	return 1;
	return {1,2,3};
	return "hello";
	return;
	return 'p';
}