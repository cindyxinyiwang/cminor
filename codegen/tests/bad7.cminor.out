x resovles to global x
type error: cannot return a boolean in a function(a) that returns integer
type error: cannot return a array [3] integer in a function(a) that returns integer
type error: cannot return a string in a function(a) that returns integer
type error: cannot return a void in a function(a) that returns integer
type error: cannot return a character in a function(a) that returns integer
